mmv1/products/iamworkforcepool/api.yaml

```
@@ -126,3 +126,195 @@ objects:
           If `sessionDuration` is not configured, minted credentials have a default duration of one hour (3600s).
           A duration in seconds with up to nine fractional digits, ending with '`s`'. Example: "`3.5s`".
         default_value: '3600s'
+  - !ruby/object:Api::Resource
+    name: 'WorkforcePoolProvider'
+    base_url: locations/{{location}}/workforcePools/{{workforce_pool_id}}/providers
+    self_link: locations/{{location}}/workforcePools/{{workforce_pool_id}}/providers/{{provider_id}}
+    create_url: locations/{{location}}/workforcePools/{{workforce_pool_id}}/providers?workforcePoolProviderId={{provider_id}}
+    update_verb: :PATCH
+    update_mask: true
+    min_version: beta
+    description: |
+      A configuration for an external identity provider.
+      ~> **Note:** Ask your Google Cloud account team to request access to workforce identity
+      federation for your billing/quota project. The account team notifies you when the project is
+      granted access.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Configure a provider within the workforce pool':
+          'https://cloud.google.com/iam/docs/manage-workforce-identity-pools-providers#configure_a_provider_within_the_workforce_pool'
+      api: 'https://cloud.google.com/iam/docs/reference/rest/v1/locations.workforcePools.providers'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'location'
+        description: The location for the resource.
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'workforcePoolId'
+        description: |
+          The ID to use for the pool, which becomes the final component of the resource name.
+          The IDs must be a globally unique string of 6 to 63 lowercase letters, digits, or hyphens.
+          It must start with a letter, and cannot have a trailing hyphen.
+          The prefix `gcp-` is reserved for use by Google, and may not be specified.
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'providerId'
+        description: |
+          The ID for the provider, which becomes the final component of the resource name.
+          This value must be 4-32 characters, and may contain the characters [a-z0-9-].
+          The prefix `gcp-` is reserved for use by Google, and may not be specified.
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          Output only. The resource name of the provider.
```

This is redundant & can be removed, but it could happen in a separate PR.

```suggestion
          The resource name of the provider.
```